### ISSUE TITLE
v2.1.0 breaks multiple arguments to Dockerfile

### ISSUE THREAD

**Issue by xf-** [docker/compose#8882]
**Description**

**Steps to reproduce the issue:**
1. Use multiple arguments 
```docker-compose.yml
version: '3.7'
services:
  #######################################
  # Application
  #######################################
  app:
    build:
      context: .docker-folder/app
      args:
        tag: 1.2.3
        container: foo
```

```Dockerfile
ARG container
ARG tag
FROM bar/${container}:${tag}
```

2. Try docker compose up

**Describe the results you received:**
Random containers tried to be pulled
 - bar/1.2.3
 - bar/foo:foo

**Describe the results you expected:**
Pull the container like in 2.0.1 (downgrade helps)

**Additional information you deem important (e.g. issue happens only occasionally):**
Looks like a race condition. Then I close the console, the result changes. (never working)

**Output of `docker compose version`:**
```
Docker Compose version 2.1.0
```

**Comment by rfay** [comment]
This is a new (and fatal) regression in 2.1.0 (and breaks ddev completely)

**Comment by rfay** [comment]
I wonder if the docker-compose team would be willing to add a rudimentary test to CI that uses ddev (as docker desktop does). ddev uses lots of features of docker-compose, and it would be lovely to see that ddev works before doing a release.

**Comment by xf-** [comment]
> I wonder if the docker-compose team would be willing to add a rudimentary test to CI that uses ddev (as docker desktop does). ddev uses lots of features of docker-compose, and it would be lovely to see that ddev works before doing a release.

Yeah, still have my other issue since v2 - `?=` vars from envs worked before v2 and now throws an error.

**Comment by ndeloof** [comment]
I ran a quick test but can't reproduce.
Are you using Buildkit or the classic builder ?
i.e. docker docker-compose build show a "blue text" build progress or old-fashionned one?

**Comment by ndeloof** [comment]
that's a significant regression indeed, I guess we will have to release v2.1.1 ASAP @ulyssessouza

**Comment by rfay** [comment]
Classic cli builder in ddev's case, because the buildx one is so noisy. 

Here's a tarball that recreates. Sounds like you've already got it though.

Please add test coverage for the build.

To use this tarball, untar it and 

```
 docker-compose -f .ssh-auth-compose-full.yaml up
[+] Running 0/1
 ⠿ ddev-ssh-agent Error                                                    3.6s
Sending build context to Docker daemon     286B
Step 1/7 : ARG BASE_IMAGE
Step 2/7 : FROM $BASE_IMAGE
1 error occurred:
	* Status: pull access denied for 1000, repository does not exist or may require 'docker login': denied: requested access to the resource is denied, Code: 1
```

[composev2build.tar.gz](https://github.com/docker/compose/files/7488684/composev2build.tar.gz)

**Comment by mat007** [comment]
I merged the linked PR as it’s clearly fixing something anyway 😬 
We should still check it fixes the repro ☝️ and add a test.

**Comment by ndeloof** [comment]
We already have https://github.com/docker/compose/blob/v2/pkg/e2e/compose_build_test.go#L44 to cover support for build args, but unfortunately

1. it only check a single arg
2. it runs with default builder, which is buildkit AFAICT. Would be nice we can run the build-related tests twice, with and without buidlkit enabled.

**Comment by ulyssessouza** [comment]
@rfay Thanks for the repro sample!
We've just released `v2.1.1` with a fix. It's still marked as `pre-release` in the release page to avoid any automatic bump on clients. 

Could you please run `ddev` tests with `v2.1.1` so we can validate that and promote that version?

**Comment by J3m5** [comment]
I tested v2.1.1 and I can say that it fixes this bug.
